### DISCUSSION TITLE
mockoon-cli list or stop commands

### DISCUSSION THREAD

**Discussion by sthuettr** [mockoon/mockoon#1344] [Help] [answered]
Hi all,
I am quite new to mocking my APIs with mockoon and wanted to integrate mockoon-cli in my gitlab-ci pipeline. For this I just wanted to execute the ```mockoon-cli start``` command, then list the running mocks with ```mockoon-cli list``` and after finishing my tests I want to stop the mock with ```mockoon-cli stop```. The ```start``` command is working, but the other two are not found, as I get an ```Command not found``` error. I found some documentation online, but the information I got was a little confusing. Sometimes, the list and stop commands are included and sometimes they are not. Now my question is, were those two commands removed in some versions or why is it that they are not found? If so, is there another way I could use mockoon-cli in my pipeline? For me docker is not really an option, because the pipeline is executed on a Windows Server machine and there, Linux containers are not supported. 
I am greatful for all the help I can get!

**Comment by 255kb** [comment, marked as answer]
Hi, 

In the past, the CLI was embarking PM2 to directly manage the running mocks. We [removed this in v4.0.0](https://mockoon.com/releases/4.0.0/#cli) together with the `stop` and `list` commands.

The CLI will now run by default as a foreground process. If you are using the Node.js ecosystem, you can [still run it using PM2](https://stackoverflow.com/a/37775318/4287789) but you will have to install it. 
On Linux it's also quite easy to run a background process by appending an `&` at the end of the command. With PowerShell you would have to use `Start-Process` (or `&` with PowerShell 6).

The CLI is also able to run multiple mocks: 
`mockoon-cli start --data ./mock1.json ./mock2.json --port 3000 3001`

The official and up-to-date documentation is [here and only here](https://github.com/mockoon/mockoon/tree/main/packages/cli#mockooncli). If you found some outdated documentation please let us know so we can update or remove it (unless it was an old blog post!).

**Reply by sthuettr** [reply]
Thank you so much! That explains it. I am simply going to use Powershell for the process management!
